tests: Change default Python command used on Windows.

stinos opened the pull request:
Default to just calling python since that is the most commonly
available: the official installer from python.org, anaconda,
msys2, chocolatey, ... all result in python being available
but not python3 (except for msys2).  On the other hand the only
system which does not have a python command by default seems to
be Cygwin, but we should not focus on what is likely the least
common usage scenario.

stinos committed:
tests: Change default Python command used on Windows.

Default to just calling python since that is most commonly
available: the official installer or zipfiles from python.org,
anaconda, nupkg all result in python being available but not
python3.  In other words: the default used so far is wrong.
Note that os.name is 'posix' when running the python version
which comes with Cygwin or MSys2 so they are not affected by
this.  However of all possible ways to get Python on Windows,
only Cygwin provides no python command so update the default
way for running tests in the README.

dlech commented:
Would it make sense to try `py -3` first? I think the default selections in the official installer install the Python launcher for Windows so most people probably have the `py` command in their path.

stinos commented:
@dlech I checked that, and as far as I can tell `py` being available is actually only with the default installer (so not on msys2 which is used quite a lot here I have the impression, also not for cygwin/conda/nupkg/zipfiles) so covers less different cases (though as you suggest, in numbers it might be bigger, hard to tell). Actually I only learned about it recently when I saw someone here use it  but that is just anecdotal :)

However if you say 'try first' you mean like `try/catch` to see if that runs? If we go there we could of course try python3/py -3/python in that order and cover everything. However that arguably is out of scope for MicroPython?

dlech commented:
Fair point. People probably aren't going to be building MicroPython on Windows with Makefiles outside of msys/cygwin anyway.

stinos committed:
windows/msvc: Use same default python command as core.

stinos commented:
Just realized: this might not even be a problem, wouldn't `os.name` on Cygwin return `posix`? Assuming it's running Cygwin's Python and not another one installed for Windows. I'll test a couple of scenarios.

stinos commented:
Results; in the 'python version' column, native means running the python version installed in the 'subsystem' e.g. for msys2 the one installed with pacman -Su python3 (using `python3` as command), whereas 'windows' means running python installed on windows itself via official installer or anaconda or similar (so, using `python` as command).

system | python version | os.name | platform.system()
--------|------------------|-----------|-------------------
cygwin | native | posix | CYGWIN_NT-10.0-19041
cygwin | windows | nt | Windows
msys2|native|posix|MSYS_NT-10.0-19041
msys2|windows|nt|Windows

I removed the extra instructions from the README again since they are not needed: apart from special cases (wanting a different version of Python for running tests than the one which is called to execute run-tests) everything regarding python selection for should work by default now.
In turn I added an extra line indicating that in the case of running everything on Cygwin, one needs to invoke python3 instead of python (well, or run in a virtual environment, or use update-alternatives, or....)

dpgeorge commented:
Thanks, looks fine to me (but I'm no expert here).  Merged in 03974485016654de09e42737ea875bc601d5ec56 and 81a4d96aed7221ea497bb90f135b3fd14ab22c0c

stinos commented:
Unless I'm really missing something this should be fine: even the v3.0 release of Python for windows had python.exe but not python3.exe.

dpgeorge commented:
I trust you here, you know better than me!  If anything is wrong I'm sure we will find out at some point.
